packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -813,6 +813,202 @@ class _TextSelectionHandleOverlayState
   }
 }
 
+/// An interface to receive selection handles from [EditableTextSelectionHandlesProvider].
+///
+/// Class implements [EditableTextGestureDetectorClient] will be able to use
+/// [EditableTextSelectionHandlesProvider].
+///
+/// See also:
+///
+///  * [EditableTextSelectionHandlesProvider], which provide basic selection handles
+///    for [EditableText].
+///  * [TextField], which implements this abstract class
+///  * [CupertinoTextField], which implements this abstract class
+abstract class EditableTextGestureDetectorClient {
+  /// Key to underlying [EditableText].
+  ///
+  /// This property will be used in retrieving [EditableTextState] and [RenderEditable]
+  /// for triggering certain selection event.
+  GlobalKey<EditableTextState> get editableTextKey;
+
+  /// Whether force press is enabled.
+  ///
+  /// If true, [EditableTextSelectionHandlesProvider] will provide handles for
+  /// force press selection.
+  bool get forcePressEnabled;
+
+  /// Whether selection is enabled.
+  ///
+  /// If false, [EditableTextSelectionHandlesProvider] will disable certain
+  /// selection handles
+  bool get selectionEnabled;
+}
+
+/// A provider that provides basic selection handles for [EditableText].
```

Comment:
What do you think about this as documentation:

Configures a [TextSelectionGestureDetector] to wrap an [EditableText].
 
The class implements sensible defaults for many user interactions with an [EditableText] (see the documentation of the various gesture handler methods, e.g. [onTapDown], [onFrocePress], etc.). Subclasses of [EditableTextSelectionHandlesProvider] can change the behavior performed in responds to these gesture events by overriding the corresponding handler methods of this class.
 
The resulting [TextSelectionGestureDetector] to wrap an [EditableText] is obtained by calling [buildGestureDetector].

See also:

 * [TextField], which uses a subclass to implement the Material-specific gesture logic of an [EditableText].
 * [CupertinoTextField], which uses a subclass to implement the Cupertino-specific gesture logic of an [EditableText].